src/price/ProgressivePrice.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+declare(strict_types=1);
+
+namespace hiqdev\php\billing\price;
+
+use hiqdev\php\billing\plan\PlanInterface;
+use hiqdev\php\billing\target\TargetInterface;
+use hiqdev\php\billing\type\TypeInterface;
+use hiqdev\php\units\Quantity;
+use hiqdev\php\units\QuantityInterface;
+use Money\Currencies\ISOCurrencies;
+use Money\Currency;
+use Money\Money;
+use Money\Parser\DecimalMoneyParser;
+
+class ProgressivePrice extends AbstractPrice
+{
+    protected ProgressivePriceThresholds $thresholds;
+
+    protected Money $price;
+
+    protected QuantityInterface $prepaid;
+
+    public function __construct(
+        $id,
+        TypeInterface $type,
+        TargetInterface $target,
+        QuantityInterface $prepaid,
+        Money $price,
+        array $thresholds,
+        ?PlanInterface $plan = null
+    ) {
+        parent::__construct($id, $type, $target, $plan);
+        $this->thresholds = new ProgressivePriceThresholds($thresholds);
+        $this->price = $price;
+        $this->prepaid = $prepaid;
+    }
+
+    /**
+     * @return ProgressivePriceThresholds
+     */
+    public function getThresholds(): ProgressivePriceThresholds
+    {
+        return $this->thresholds;
+    }
+
+    public function getPrepaid(): QuantityInterface
+    {
+        return $this->prepaid;
+    }
+
+    public function getPrice(): Money
+    {
+        return $this->price;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function calculateUsage(QuantityInterface $quantity): ?QuantityInterface
+    {
+        $usage = $quantity->convert($this->prepaid->getUnit());
+
+        if ($usage->isPositive()) {
+            return $usage;
+        }
+
+        return Quantity::create($this->prepaid->getUnit()->getName(), $quantity->getQuantity());
```

Comment:
What is a sense of creating a Quantity for a negative value? To giveaway money to customers?